### ISSUE TITLE
Past answers produce a warning when using problem groups

### ISSUE THREAD

**Issue by dlglin** [openwebwork/webwork2#1850]
In 2.17 when viewing past answers on a GW quiz which uses problem groups the user gets a warning message.

The offending code seems to be https://github.com/openwebwork/webwork2/blob/073e9f5886a4a73fbff92d4c9a835ca18e1d367b/lib/WeBWorK/ContentGenerator/Instructor/ShowAnswers.pm#L138:142

This gets a merged set and problem based on the unversioned set, which means that the source file could be something like "group:groupname", which then fails because this is not a valid source file.

In develop the warning is not present, but the behaviour is still not what is expected.  The new logic skips the generation of the "silly problem" since the merged problem was not created, which means that the answer type for that problem is not stored.  When displaying the past answers the logic tries to check the answer type to see if it is an essay problem, and since the `$answerType` variable will be empty it will not get the correct result if the question is indeed an essay problem.

**Comment by drgrice1** [comment]
I see the warning with 2.17.  I don't see the warning on the develop branch.  I am not sure why though.  That code hasn't changed since before 2.17 other than clean up (perltidy and spacing).  Where is the new logic that you are talking about?  It isn't in the lines you referenced.

I didn't test with an essay problem.  I don't have many of those to begin with, and none in a quiz pool problem.  It will take a little work to set up a test for that.

**Comment by dlglin** [comment]
Sorry, the logic is the same.  I misread it.  I'll have to look more carefully to figure out why there's no warning in develop.

In 2.17 the PG object is generated by `WeBWorK::PG->new`, whereas in develop it's `renderPG`.  Is it possible that `renderPG` just isn't spitting the warnings?

**Comment by drgrice1** [comment]
So the reason that there is no warning is because of the new way that warnings and errors are dealt with.  Since pg problem rendering occurs in a subprocess the warnings do not automatically matriculate back to the main process.  Instead they are caught in the subprocess, and sent back with the rest of the pg problem data.  In Problem.pm, GatwayQuiz.pm, and some of the other places that problem rendering occurs those warnings and errors are dealt with.  In this instance I decided not to re-warn the warnings.  I don't think they are really pertinent in this situation.  Errors were not dealt with before in this case, so I didn't add that.  This is actually an error now, and not a warning it seems.  Probably with the restructuring of the translator. 

In any case, I can verify that the problem is still being processed and failing because the grouping problem is not a valid file name.  I am certain that grouping problems never worked for this, and this is an issue that goes back further than 2.17.  It also seems that processing the problem is entirely unnecessary here also.  The essay flag is saved in the database.  Although it is only saved once a student (any one student) has answered it, but that is the only case that it matters here anyway.

**Comment by drgrice1** [comment]
Hmm, so to complicate matters, Gateway quizzes never save the essay flag to the database.

**Comment by dlglin** [comment]
When I first saw the warning I did wonder why problems were being processed on the past answers page.  It definitely seems like overkill to process the problem just to determine if it has essay answers.

I'll note that we don't actually use essay problems in quizzes here.  After I saw the warning I dug a little to see if it actually broke anything, and it appears to only affect the past answers page for versioned sets with essay answers.

Also it's unclear to me why the set is being unversioned before getting the merged problem.  It seems like getting the merged problem for the version should work, since that will have a valid source file.
